### PR TITLE
feat: implement profile view

### PR DESCRIPTION
**Pull request by aman-singh7**

### CODE REVIEW

File: lib/presentation/update_profile/bloc/update_profile_bloc.dart [hunk 1]
```diff
@@ -0,0 +1,253 @@
+import 'dart:async';
+import 'dart:io';
```
Comment (BURG3R5) [inline lib/presentation/update_profile/bloc/update_profile_bloc.dart:2]: Are these imports being used?

Comment (BURG3R5) [inline lib/presentation/update_profile/bloc/update_profile_bloc.dart:2]: Are these imports being used?

File: lib/presentation/profile/bloc/profile_bloc.dart [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+  }
+
+  // Cell Color for Acceptance Graph
+  Color getCellColor(DateTime date) {
+    final index = _activity[date] ?? 0;
+    if (index < 0) {
+      return Color.fromRGBO(255, 0, 0, math.min(-0.2 * index, 1));
+    } else if (index > 0) {
+      return Color.fromRGBO(0, 255, 0, math.min(0.2 * index, 1));
+    }
```
Comment (BURG3R5) [inline lib/presentation/profile/bloc/profile_bloc.dart:160]: Can't you just use `math.abs` on the index?

Comment (BURG3R5) [inline lib/presentation/profile/bloc/profile_bloc.dart:160]: Can't you just use `math.abs` on the index?

**Review by BURG3R5 — CHANGES_REQUESTED** [review]

File: lib/presentation/profile/bloc/profile_bloc.dart (hunk 2 above)
Comment (aman-singh7) [reply lib/presentation/profile/bloc/profile_bloc.dart:160]: See the RGB values; they are different.

Comment (aman-singh7) [reply lib/presentation/profile/bloc/profile_bloc.dart:160]: Refer above.

File: lib/presentation/update_profile/bloc/update_profile_bloc.dart (hunk 1 above)
Comment (aman-singh7) [reply lib/presentation/update_profile/bloc/update_profile_bloc.dart:2]: Yup

File: lib/presentation/update_profile/widgets/profile_details.dart [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+      children: [
+        Padding(
+          padding: EdgeInsets.symmetric(vertical: 16.h),
+          child: _profilePicture(),
+        ),
+        BlocBuilder<UpdateProfileBloc, UpdateProfileState>(
+          builder: (context, state) {
+            return TextInput(
+              controller: UpdateProfileBloc.controllers['name'],
+              onChanged: (val) {},
```
[the remark this answers is not in the document]
Comment (BURG3R5) [reply lib/presentation/update_profile/widgets/profile_details.dart:28]: @aman-singh7 Can you check this out?

File: lib/presentation/profile/widgets/submission_display.dart [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+  final String submissionType;
+  final int value;
+  final String key;
+}
+
+extension on SubmissionData {
+  double percentage(int total) {
+    return double.parse((value * 100 / total).toStringAsFixed(1));
+  }
```
[the remark this answers is not in the document]
Comment (BURG3R5) [reply lib/presentation/profile/widgets/submission_display.dart:149]: @aman-singh7 Can you check this out? I think these suggestions were hidden before.

**Review by BURG3R5 — APPROVED** [review]

**Pull request merged**